server/controllers/stock/index.js

```diff
@@ -975,3 +977,221 @@ function getStockTransfers(req, res, next) {
+          lot_uuid : db.bid(lot.uuid),
+          depot_uuid : db.bid(movement.depot_uuid),
+          document_uuid : db.bid(lossUuid),
+          quantity : lot.quantity_lost,
+          unit_cost : lot.unit_cost,
+          date : new Date(movementDate),
+          is_exit : 1,
+          flux_id : core.flux.TO_LOSS,
+          description : movement.description,
+          user_id : req.session.user.id,
+          period_id : periodId,
+        };
+        trx.addQuery('INSERT INTO stock_movement SET ?', lossMovementObject);
+      });
+
+      const stockConsumptionParams = [
+        db.bid(consumptionUuid), 1, req.session.project.id, req.session.enterprise.currency_id,
+      ];
+
+      const stockLossParams = [
+        db.bid(lossUuid), 1, req.session.project.id, req.session.enterprise.currency_id,
+      ];
+
+      if (req.session.stock_settings.enable_auto_stock_accounting) {
+        if (stockConsumptionQuantities.length > 0) {
+          trx.addQuery('CALL PostStockMovement(?)', [stockConsumptionParams]);
+        }
+
+        if (stockLossQuantities.length > 0) {
+          trx.addQuery('CALL PostStockMovement(?)', [stockLossParams]);
+        }
+      }
+    });
+
+    consumptionDetailled.forEach(item => {
+      item.detailled.forEach(elt => {
+        const consumptionUuid = uuid();
+        const lossUuid = uuid();
+
+        let eltDate = new Date(elt.end_date);
+        eltDate = eltDate.setHours(23, 30, 0);
+
+        if (elt.quantity_consumed > 0) {
+          const consumptionMovementObject = {
+            uuid : db.bid(uuid()),
+            lot_uuid : db.bid(item.uuid),
+            depot_uuid : db.bid(movement.depot_uuid),
+            document_uuid : db.bid(consumptionUuid),
+            quantity : elt.quantity_consumed,
+            unit_cost : item.unit_cost,
+            date : new Date(eltDate),
+            is_exit : 1,
+            flux_id : core.flux.AGGREGATE_CONSUMPTION,
+            description : movement.description,
+            user_id : req.session.user.id,
+            period_id : periodId,
+          };
+          trx.addQuery('INSERT INTO stock_movement SET ?', consumptionMovementObject);
+
+          const consumptionParams = [
+            db.bid(item.inventory_uuid),
+            db.bid(movement.depot_uuid), new Date(`${elt.end_date} 23:30`), elt.quantity_consumed,
+          ];
+
+          trx.addQuery('CALL ComputeStockConsumptionByDate(?, ?, ?, ?)', consumptionParams);
+        }
+
+        if (elt.quantity_lost > 0) {
+          const lossMovementObject = {
+            uuid : db.bid(uuid()),
+            lot_uuid : db.bid(item.uuid),
+            depot_uuid : db.bid(movement.depot_uuid),
+            document_uuid : db.bid(lossUuid),
+            quantity : elt.quantity_lost,
+            unit_cost : item.unit_cost,
+            date : new Date(eltDate),
+            is_exit : 1,
+            flux_id : core.flux.TO_LOSS,
+            description : movement.description,
```

Comment:
We should do what we do with the other stock movements and template in our own description as well.  Something like "Aggregated Stock Adjustment for Period X by User Y"